### PR TITLE
Implementation of detailed execution plan

### PR DESCRIPTION
**Pull request by vasiliy-yashkov**

### CODE REVIEW

File: src/main/org/firebirdsql/gds/ng/FbStatement.java [hunk 1]
```diff
@@ -227,6 +227,13 @@
      */
     String getExecutionPlan() throws SQLException;
 
+    /**
+     * @return The detailed execution plan of the currently prepared statement
+     * @throws SQLException
+     *         If this statement is closed.
+     */
+    String getExplainedExecutionPlan() throws SQLException;
```
Comment (mrotteveel) [inline src/main/org/firebirdsql/gds/ng/FbStatement.java]: Although it matches the name in the Firebird 3 release notes, I'm not sure I like the name 'explained plan'. I might change it after accepting the merge.

It's also possible that I'll change this to a method `getExecutionPlan(PlanType)` where `PlanType` would be an enum that has both types.

NOTE: Just observation, no need to change this.

**Comment by mrotteveel** [comment]
I'll take another look later this week as a sanity check. 

The only point that may need addressing is the question if a version check + exception might be needed, as I'm not sure if Firebird 2.5 and lower will return an understandable exception message in this case.

**Comment by mrotteveel** [comment]
I've taken another look, no new things found.

File: src/main/org/firebirdsql/gds/ng/AbstractFbStatement.java [hunk 2]
```diff
@@ -416,6 +416,12 @@ public final String getExecutionPlan() throws SQLException {
         return getSqlInfo(processor.getDescribePlanInfoItems(), getDefaultSqlInfoSize(), processor);
     }
 
+    @Override
+    public final String getExplainedExecutionPlan() throws SQLException {
+        final ExecutionPlanProcessor processor = createExecutionPlanProcessor();
```
[the remark this answers is not in the document]
Comment (mrotteveel) [reply src/main/org/firebirdsql/gds/ng/AbstractFbStatement.java]: Implemented on a local branch, will merge shortly.

**Comment by mrotteveel** [comment]
I will merge this after my final tests have run. 

One thing I noticed is that you implemented this on branch master, which resulted in your pull request including a merge commit (ab57730) from upstream/master. I suggest that you leave 'master' untouched (and only keep it in-sync with Jaybird master), and implement changes on separate branches: that should yield cleaner merge requests.

**Comment by vasiliy-yashkov** [comment]
Ok, I get it. Thanks, Mark!

**Comment by mrotteveel** [comment]
Shadow ticket: http://tracker.firebirdsql.org/browse/JDBC-574

**Comment by mrotteveel** [comment]
Manually closing this, as I merged it locally with a rebased branch. Thanks for your contribution!

**Pull request closed without merging**